### PR TITLE
LoggerState: test cases

### PR DESCRIPTION
**Pull request by deanlee**
multiple threaded logging problem during testing that needs to be fixed:
- [x] SentinelType may not be the last msg in the log

### CODE REVIEW

**Comment by pd0wm** [comment]
What's the benefit over the python test?

**Comment by deanlee** [comment]
it can do functional testing

**Comment by adeebshihadeh** [comment]
I'd like this to supplement the python tests. Let's keep the existing python test cases.

**Comment by deanlee** [comment]
closed. move to c++ logger #21877

File: selfdrive/loggerd/logger.h [hunk 1]
```diff
@@ -53,8 +54,12 @@ class BZFile {
   BZFILE* bz_file = nullptr;
 };
 
+typedef cereal::Sentinel::SentinelType SentinelType;
+
 typedef struct LoggerHandle {
   pthread_mutex_t lock;
+  SentinelType end_sentinel_type;
+  bool exit_signal;
```
Comment (pd0wm) [inline selfdrive/loggerd/logger.h:62]: exit signal is not a bool. We want to distinguish SIGINT from SIGPWR.

Comment (deanlee) [reply selfdrive/loggerd/logger.h]: fixed

**Pull request merged**